pkg/providers/vsphere/vmprovider_vm.go

```diff
@@ -1486,16 +1488,188 @@ func (vs *vSphereVMProvider) getVMYamlFromSnapshot(
+		} else {
+			// Taking only the first one for now
+			// TODO(nabarun): Once consensus is established on what would be the storage class when multiple storage
+			// policies are returned, update this logic if needed
+			profileID := profileIDs[0]
+			storageClasses, err := spq.GetStorageClassesForPolicy(vmCtx, vs.k8sClient, vmCtx.VM.Namespace, profileID.UniqueId)
+			if err != nil {
+				vmCtx.Logger.Error(err, "error finding storage classes")
+				return "", err
+			}
+
+			if len(storageClasses) == 0 {
+				err := errors.New("error finding storage classes for profile")
+				vmCtx.Logger.Error(err, "", "profileId", profileID)
+				return "", err
+			}
+
+			// in SV, it seems there are two storage classes for the same profileId
+			// TODO(nabarun): Reach consensus on which one to pick up. Currently picking up the first one found.
+			if len(storageClasses) > 1 {
+				vm.Spec.StorageClass = storageClasses[0].Name
+			}
+		}
+
+		// Inferred from the power state of the snapshot
+		// It doesn't matter what the power state of the VM is at the time of revert.
+		// The VM is always reverted in the same power state as the snapshot.
+		snapshot := vmlifecycle.FindSnapshotInTree(vmCtx.MoVM.Snapshot.RootSnapshotList, snap.Value)
+		if snapshot == nil {
+			// weird situation here
+			// the snapshot being processed doesn't exist in the snapshot tree
+			// this shouldn't be happening
+			err := fmt.Errorf("snapshot %s not found in VM Snapshot Tree", snap.Value)
+			vmCtx.Logger.Error(err, "")
+			return "", err
+		}
+		// convert from vimtypes.VirtualMachinePowerState to vmopv1.VirtualMachinePowerState
+		poweredOn := snapshot.State == vimtypes.VirtualMachinePowerStatePoweredOn
+		if poweredOn {
+			vm.Spec.PowerState = vmopv1.VirtualMachinePowerStateOn
+		} else {
+			vm.Spec.PowerState = vmopv1.VirtualMachinePowerStateOff
+		}
+
+		// Number of elements in the spec.network.interfaces list must be same as number of VM's NICs from ConfigInfo
+		// Details of each interface (which network does it connect to) is ignored
+		// Networking can be fixed after revert and it is not possible to map the vSphere network to Supervisor networks.
+		devices := object.VirtualDeviceList(moSnap.Config.Hardware.Device)
+		vNICs := devices.SelectByType((*vimtypes.VirtualEthernetCard)(nil))
+		if len(vNICs) == 0 {
+			// If no vNICs are defined, then disable network
+			vm.Spec.Network.Disabled = true
+		} else {
+			interfaces := make([]vmopv1.VirtualMachineNetworkInterfaceSpec, len(vNICs))
+			for i := range vNICs {
+				interfaces[i] = vmopv1.VirtualMachineNetworkInterfaceSpec{
+					Name: fmt.Sprintf("eth-%d", i),
+				}
+			}
+			vm.Spec.Network.Interfaces = interfaces
+		}
+
+		// MinHardwareVersion is set to the Current Hardware Version of the VM
+		chv, err := strconv.ParseInt(moSnap.Config.Version, 10, 32)
+		if err != nil {
+			vmCtx.Logger.V(5).Info("unable to approximate minimum hardware version", "managed object hardware version", moSnap.Config.Version)
```

Comment:
We follow a convention of making the log markers camel cased so it's easier to search through.  So, consider calling it `hwVersionFromSnapshot` instead of `managed object hardware version`